### PR TITLE
testing! (POC for UDT)

### PR DESCRIPTION
**Pull request by himanshukohli09**

### CODE REVIEW

File: connector/third_party/apache-spark/src/main/java/com/google/cloud/spark/bigquery/ArrowSchemaConverter.java [hunk 1]
```diff
@@ -152,7 +154,7 @@ private static DataType fromArrowType(ArrowType arrowType)
       case Date: return DataTypes.DateType;
       case Time:
       case Timestamp: return DataTypes.TimestampType;
-      case Decimal: return DataTypes.createDecimalType();
```
Comment (davidrabinowitz) [inline connector/third_party/apache-spark/src/main/java/com/google/cloud/spark/bigquery/ArrowSchemaConverter.java:155]: Why to remove the decimal type? can' we distinct between the two?

File: connector/third_party/apache-spark/src/main/java/com/google/cloud/spark/bigquery/ArrowSchemaConverter.java [hunk 2]
```diff
@@ -370,6 +374,25 @@ final Decimal getDecimal(int rowId, int precision, int scale) {
...
+    Decimal256Accessor(Decimal256Vector vector) {
+      super(vector);
+      this.accessor = vector;
+    }
+
+    UTF8String getUTF8String(int rowId){
+      if (isNullAt(rowId)) return null;
+      BigDecimal bd = accessor.getObject(rowId);
+      UTF8String utf8String  = UTF8String.fromString(bd.toPlainString());
+      System.out.println("ARROW => " + utf8String.toString());
```
Comment (davidrabinowitz) [inline connector/third_party/apache-spark/src/main/java/com/google/cloud/spark/bigquery/ArrowSchemaConverter.java:390]: Please remove System.out.println()

File: connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java [hunk 3]
```diff
@@ -148,10 +149,17 @@ static Object convertByBigQueryType(Field field, Object value) {
...
       Decimal d = Decimal.apply(b, BQ_NUMERIC_PRECISION, BQ_NUMERIC_SCALE);
-
       return d;
     }
 
+    if(LegacySQLTypeName.BIGNUMERIC.equals(field.getType())){
+      byte[] bytes = getBytes( (ByteBuffer) value);
+      BigDecimal b = new BigDecimal(new BigInteger(bytes), 38);
+      UTF8String str = UTF8String.fromString(b.toString());
+      System.out.println("AVRO:   " +  str);
```
Comment (davidrabinowitz) [inline connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java:159]: Please remove System.out.println()

File: connector/third_party/apache-spark/src/main/java/org/apache/spark/bigquery/BigNumericUDT.scala [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    val bigNumeric = new BigNumeric(new BigDecimal(utf8str.toString))
+    System.out.println("deserialize:   " + utf8str.toString)
+    bigNumeric
+  }
+
+  override def userClass: Class[BigNumeric] = classOf[BigNumeric]
+}
+
+@SQLUserDefinedType(udt = classOf[BigNumericUDT])
+class BigNumeric(val number: BigDecimal){
```
Comment (davidrabinowitz) [inline connector/third_party/apache-spark/src/main/java/org/apache/spark/bigquery/BigNumericUDT.scala:31]: Can be a case class

File: connector/third_party/apache-spark/src/main/java/org/apache/spark/bigquery/BigNumericUDT.scala [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.math.BigDecimal
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow
+import org.apache.spark.sql.types._
+import org.apache.spark.unsafe.types.UTF8String
+
+private[spark] class BigNumericUDT extends UserDefinedType[BigNumeric]{
+  override def sqlType: DataType = StringType //DataTypes.createDecimalType
```
Comment (davidrabinowitz) [inline connector/third_party/apache-spark/src/main/java/org/apache/spark/bigquery/BigNumericUDT.scala:11]: Why not BigNumericType ?

File: connector/third_party/apache-spark/src/main/java/org/apache/spark/bigquery/BigNumericUDT.scala [hunk 6]
```diff
@@ -0,0 +1,32 @@
+package org.apache.spark.bigquery
+
+import java.math.BigDecimal
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow
+import org.apache.spark.sql.types._
+import org.apache.spark.unsafe.types.UTF8String
+
+private[spark] class BigNumericUDT extends UserDefinedType[BigNumeric]{
```
Comment (davidrabinowitz) [inline connector/third_party/apache-spark/src/main/java/org/apache/spark/bigquery/BigNumericUDT.scala:10]: What about python support?

**Review by himanshukohli09 — COMMENTED** [review]
test

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
Looks like a good start. Can you you please also check if all the code can be developed in Java for our upcoming change?

**Review by davidrabinowitz — COMMENTED** [review]
Please extend the all_types test table to include bignumeric as well

**Pull request closed without merging**